csharp/src/Google.Protobuf.Test/Collections/RepeatedFieldTest.cs

```diff
@@ -303,6 +303,19 @@ public void Enumerator()
             }
         }
 
+        [Test]
+        public void Throw_InvalidOperation_Exception_if_modified()
+        {
+            var list = new RepeatedField<string> { "first", "second" };
+            using (var enumerator = list.GetEnumerator())
+            {
+                list.RemoveAt(0);
```

Comment:
What if remove one item and then add one item back during enumeration operation?

Reply:
it's possible but unlikely. I just covered the most common scenario to limit the footprint of the patch. 

A better approach would be to clone the internal array and enumerate the snapshot but will impact memory usage. 80% of the time you'll enumerate to consume repeated field.

An even safer approach would be adding checks on array mutations while an enumeration is in progress.

Reply:
Does "add checks on array mutations while enumeration is in progress" cost lot? If no can you change to this way?